sentry-core/src/main/java/io/sentry/core/SentryEvent.java

```diff
@@ -203,30 +203,58 @@ public void setFingerprint(List<String> fingerprint) {
     this.fingerprint = fingerprint;
   }
 
+  public void addFingerprint(String fingerprint) {
+    if (this.fingerprint == null) {
+      this.fingerprint = new ArrayList<>();
+    }
+    this.fingerprint.add(fingerprint);
+  }
+
   public List<Breadcrumb> getBreadcrumbs() {
     return breadcrumbs;
   }
 
-  public void setBreadcrumbs(ArrayList<Breadcrumb> breadcrumbs) {
+  public void setBreadcrumbs(List<Breadcrumb> breadcrumbs) {
     this.breadcrumbs = breadcrumbs;
   }
 
+  public void addBreadcrumb(Breadcrumb breadcrumb) {
+    if (breadcrumbs == null) {
+      breadcrumbs = new ArrayList<>();
+    }
+    breadcrumbs.add(breadcrumb);
+  }
+
   public Map<String, String> getTags() {
     return tags;
   }
 
-  public void setTags(HashMap<String, String> tags) {
+  public void setTags(Map<String, String> tags) {
     this.tags = tags;
   }
 
+  public void addTag(String key, String value) {
```

Comment:
Unified API calls it `setTag` since it's a dictionaru and will override an existent tag with the same key

```suggestion
  public void setTag(String key, String value) {
```